### PR TITLE
Adding e2e scenarios for devfile

### PR DESCRIPTION
**Pull request by rnapoles-rh**
**What type of PR is this?**
/kind feature

**What does does this PR do / why we need it**:
Adding new end-to-end tests for devfile debug support

**Which issue(s) this PR fixes**:

Fixes part of #3377 

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
make test-e2e-devfile

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -235,21 +235,11 @@ test-cmd-devfile-exec:
 test-cmd-docker-devfile-exec:
 	ginkgo $(GINKGO_FLAGS) -focus="odo docker devfile exec command tests" tests/integration/devfile/docker
 
-# Run odo status devfile command tests
-.PHONY: test-cmd-devfile-status
-test-cmd-devfile-status:
-	ginkgo $(GINKGO_FLAGS) -focus="odo devfile status command tests" tests/integration/devfile/
-
```
Comment (mik-dass) [inline Makefile:242]: why are we removing the status integration tests?

File: Makefile [hunk 2]
```diff
@@ -235,21 +235,11 @@ test-cmd-devfile-exec:
...
 # Run odo devfile watch command tests
 .PHONY: test-cmd-devfile-watch
 test-cmd-devfile-watch:
 	ginkgo $(GINKGO_FLAGS) -focus="odo devfile watch command tests" tests/integration/devfile/
 
-# Run odo devfile app command tests
-.PHONY: test-cmd-devfile-app
```
Comment (mik-dass) [inline Makefile:249]: why are we removing the app integration tests?

File: Makefile (hunk 1 above)
Comment (rnapoles-rh) [reply Makefile:242]: I am not sure why it is marked for deletion. My change was to add test-e2e-devfile .

Could it be that I forgot to run a checkout from master before starting my changes?

File: Makefile (hunk 2 above)
Comment (rnapoles-rh) [reply Makefile:249]: same as above. 
How should I proceed to get them back in? 
Can we cancel this PR and I create a new one?

Comment (mik-dass) [reply Makefile:249]: Fetching the latest master and rebasing might help.

File: Makefile (hunk 1 above)
Comment (mik-dass) [reply Makefile:242]: The app and status tests were newly added and maybe that's why they are shown removed.

**Comment by rnapoles-rh** [comment]
Will rebase and add the missing tests to the Makefile

**Pull request closed without merging**